### ISSUE TITLE
Feature request: follow symlinks like ls -L, --dereference, or be even friendlier

### ISSUE THREAD

**Issue by tv42** [ogham/exa#393]
Hi. I have some large files managed externally, and symlinks to them in git. I often want to see the sizes of the large files, not the sizes of the symlinks.

In fact, I consider displaying the size of the *symlink* an exercise in silliness; I'd much rather see the size of the target.

Exa doesn't seem to have anything like `ls -L` to give me that information.

1. Could something like `ls -L` be added?
2. Maybe even just always fetch the target size when a symlink size is asked for.

**Comment by erkin** [comment]
Yes! `--dereference-command-line-symlink-to-dir` is the only feature of ls I'm missing.
When I ls a symlink to a dir, I simply want to see inside of what it's pointing at, not the filename.

**Comment by pronobis** [comment]
Would be great to have that feature. I like sorting directories first, and links to directories do not get placed on top without this option.

**Comment by Svadkos** [comment]
Also would be great if there was symlink indicator even with this option toggled.

**Comment by lilyball** [comment]
I would really like to see this. `exa` seems to behave like `ls -H` already in most cases (except when using `-T`, filed as #549), but an equivalent to `-L` would be useful, in particular with `-T` (though with `-T` it would need some kind of loop detection).

**Comment by sklages** [comment]
It's been around for years now?? Wow .. okay, just another guy interested in `-L` (and maybe`-H`).

This is a quite fundamental parameter of `ls` .. I am a bit puzzled that this has not been implemented from the very beginning ...

**Comment by adam-zethraeus** [comment]
+1

**Comment by Legisign** [comment]
> Yes! `--dereference-command-line-symlink-to-dir` is the only feature of ls I'm missing.
> When I ls a symlink to a dir, I simply want to see inside of what it's pointing at, not the filename.

I wholeheartedly agree! If a symlink is to a dir, it’s for a reason.

**Comment by Jaakkonen** [comment]
Using the current `eza --tree --follow-symlinks` works quite well but has issues in kernel `/sys` directory with a bunch of back-symlinks. I'd preferably have it possible to limit the depth that the symlinks are followed independently from the normal `--tree` `-L` option. The depth is already required for the tree command to not get stuck in infinite loop but also pollutes the output quite a bit due to re-expanding the backlinks continuously
